### ISSUE TITLE
Request is reaching forwarder with empty Payload

### ISSUE THREAD

**Issue by artem-belov** [networkservicemesh/sdk#788]
Steps to reproduce:
[deployments-k8s](https://github.com/networkservicemesh/deployments-k8s)
Commit 9f8c70478980b3d09c04158229495ba53034da6d
Memif2Vxlan2Memif example

Forwarder does not assign VXLAN mechanism because connection `Payload` is empty

```
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(25)                         ⎆ sdk-vpp/pkg/networkservice/mechanisms/memif/memifClient.Request()
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(25.1)                           request-diff={"mechanism_preferences":{"+0":{"cls":"LOCAL","type":"MEMIF"}}}
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(26)                          ⎆ sdk-vpp/pkg/networkservice/mechanisms/memif/memifproxy/memifProxyClient.Request()
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(27)                           ⎆ sdk-vpp/pkg/networkservice/mechanisms/kernel/kerneltap/kernelTapClient.Request()
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(27.1)                             request-diff={"mechanism_preferences":{"+1":{"cls":"LOCAL","type":"KERNEL"}}}
Mar 24 12:41:34.841[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(28)                            ⎆ sdk/pkg/networkservice/core/next/nextClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(29)                             ⎆ sdk-vpp/pkg/networkservice/mechanisms/vxlan/vxlanClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(30)                              ⎆ sdk-vpp/pkg/networkservice/mechanisms/vxlan/vni/vniClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(31)                               ⎆ sdk/pkg/networkservice/common/mechanisms/recvfd/recvFDClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(32)                                ⎆ sdk/pkg/networkservice/common/mechanisms/sendfd/sendFDClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(33)                                 ⎆ sdk/pkg/networkservice/common/null/nullClient.Request()
Mar 24 12:41:34.842[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(34)                                  ⎆ api/pkg/api/networkservice/networkServiceClient.Request()
Mar 24 12:41:34.913[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(34.1)                                    response={"id":"ed03f0c0-2919-4c71-ab7d-892cda446f5b","network_service":"icmp-responder","mechanism":{"cls":"LOCAL","type":"MEMIF","parameters":{"inodeURL":"inode://2097182/3291948"}},"context":{"ip_context":{"src_ip_addr":"172.16.1.101/32","dst_ip_addr":"172.16.1.100/32","src_routes":[{"prefix":"172.16.1.100/32"}],"dst_routes":[{"prefix":"172.16.1.101/32"}]}},"path":{"index":5,"path_segments":[{"name":"nsc-64f57dc45f-lp6vl","id":"nsc-64f57dc45f-lp6vl-0","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTMyODcsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zLXJmZGx6L3NhL2RlZmF1bHQifQ.2uhdt3dCw5IvSZXkjFh6Z9KR4SSiGaqo3npV3hPwrKtKz0FEVlz3LQPCNmQhv9bp4FGGhZcuhNt2c24YvkjN5A","expires":{"seconds":1616592653}},{"name":"nsmgr-2rgzq","id":"d65ce174-3c6c-4e6a-9082-7d71ac05a13f","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTMsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.pNUlVQ4y0Mn2JM6JRK0dl2LtLdVzR0aE_FE7pFwiLgmxvj-eq7WRJ1GEvW5cvCkMz9ZwbhENvathlaSZEsrGeg","expires":{"seconds":1616592653}},{"name":"forwarder-vpp-mnsmq","id":"70d75816-8708-4b5d-89c6-d8db3e27758c","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTMsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.EyNzv1AyTpBiILgQU2BzA06xG_mMhtZzsGlz5dRZHX5_tJCV9WEh16WPC74qFQ9UbfU26q1LUiuOxdjaHlGQ2A","expires":{"seconds":1616592653}},{"name":"nsmgr-2rgzq","id":"33ff1166-a5e5-4e7e-aa95-d08afc7c6776","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTMsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.9p42tgSCcjPLz9t-0xCTsRd86x_-O9GP0XJIIVAxb30AKIbNsqo3k8iCkD1M1UFiMiGOUOWXDETudOGPE_HFmw","expires":{"seconds":1616592651}},{"name":"nsmgr-2nsjs","id":"1844cfd6-60ab-4f83-a0a0-6cf1bbf0ae0d","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTEsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.gyKiaw48BJH_D1O_NSu4xQFINzaQGDBxwfq8H-UHTSA-fS4e234_3CpPhG2ogzaBGo1rFisVEBm4YdYlKaeeOw","expires":{"seconds":1616592651}},{"name":"forwarder-vpp-2rx2r","id":"ed03f0c0-2919-4c71-ab7d-892cda446f5b","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTEsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.6BYtpsyGVUsNsbLKj8TITyb9Pgh3etcHhT_-4AH4T2GeOXuIL54zDf_FfMgFY6K1wDSbfDFMfix-r4VD5xnDQg","expires":{"seconds":1616592651}},{"name":"nsmgr-2nsjs","id":"50c4bc7e-853d-4519-9577-ab1a7b93c407","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9ucy1yZmRsei9zYS9kZWZhdWx0IiwiZXhwIjoxNjE2NTkyNjUxLCJzdWIiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQifQ.Cu14XrJby_wC_6x-XnD28qZdlUpMRJ5bbmgwx8imv57ht4BjbVxXAlM1k78I3yoUEs1EzbVpQYp-SQfYjqiAYA","expires":{"seconds":1616592651}},{"name":"nse-78ffc69bfd-kschb","id":"3ed638f3-3598-4611-b6df-f88adcf08b64","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTMyODgsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zLXJmZGx6L3NhL2RlZmF1bHQifQ.1cVVGMJSUGwj9dsBXEKY0-sRNhZH5Xx9FynTyROuv6R4FkpyUeLHmqv3yYJovsiryV6P92trv6yZ6WOWAqSY4Q","expires":{"seconds":1616592651}}]},"network_service_endpoint_name":"nse-78ffc69bfd-kschb"}
Mar 24 12:41:34.917[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(31.1)                                 response-diff={"mechanism":{"parameters":{"inodeURL":"file:///proc/1/fd/22"}}}
Mar 24 12:41:34.919[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(26.1)                            response-diff={"mechanism":{"parameters":{"inodeURL":"file:///tmp/memifproxy/ed03f0c0-2919-4c71-ab7d-892cda446f5b/memif.socket"}}}
Mar 24 12:41:34.928[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(22.1)                        response-diff={"path":{"path_segments":{"5":{"metrics":{"+client_drops":"0","+client_rx_bytes":"0","+client_rx_packets":"0","+client_tx_bytes":"0","+client_tx_packets":"0"}}}}}
Mar 24 12:41:34.929[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(18.1)                    response-diff={"mechanism":{"cls":"","parameters":{"-inodeURL":"file:///tmp/memifproxy/ed03f0c0-2919-4c71-ab7d-892cda446f5b/memif.socket"},"type":""}}
Mar 24 12:41:34.931[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(12.1)              request-diff={"connection":{"context":{"ip_context":{"dst_ip_addr":"172.16.1.100/32","dst_routes":{"+0":{"prefix":"172.16.1.101/32"}},"src_ip_addr":"172.16.1.101/32","src_routes":{"+0":{"prefix":"172.16.1.100/32"}}}},"path":{"path_segments":{"+6":{"name":"nsmgr-2nsjs","id":"50c4bc7e-853d-4519-9577-ab1a7b93c407","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9ucy1yZmRsei9zYS9kZWZhdWx0IiwiZXhwIjoxNjE2NTkyNjUxLCJzdWIiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQifQ.Cu14XrJby_wC_6x-XnD28qZdlUpMRJ5bbmgwx8imv57ht4BjbVxXAlM1k78I3yoUEs1EzbVpQYp-SQfYjqiAYA","expires":{"seconds":1616592651}},"+7":{"name":"nse-78ffc69bfd-kschb","id":"3ed638f3-3598-4611-b6df-f88adcf08b64","token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTMyODgsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zLXJmZGx6L3NhL2RlZmF1bHQifQ.1cVVGMJSUGwj9dsBXEKY0-sRNhZH5Xx9FynTyROuv6R4FkpyUeLHmqv3yYJovsiryV6P92trv6yZ6WOWAqSY4Q","expires":{"seconds":1616592651}},"5":{"metrics":{"+client_drops":"0","+client_rx_bytes":"0","+client_rx_packets":"0","+client_tx_bytes":"0","+client_tx_packets":"0"},"token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJzcGlmZmU6Ly9leGFtcGxlLm9yZy9ucy9uc20tc3lzdGVtL3NhL2RlZmF1bHQiLCJleHAiOjE2MTY1OTI2NTEsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL25zbS1zeXN0ZW0vc2EvZGVmYXVsdCJ9.6BYtpsyGVUsNsbLKj8TITyb9Pgh3etcHhT_-4AH4T2GeOXuIL54zDf_FfMgFY6K1wDSbfDFMfix-r4VD5xnDQg"}}}},"mechanism_preferences":{"-0":{"cls":"LOCAL","type":"MEMIF"},"-1":{"cls":"LOCAL","type":"KERNEL"}}}
Mar 24 12:41:34.931[36m [INFO] [name:forwarder-vpp-2rx2r] [0m(13)             ⎆ sdk/pkg/networkservice/common/mechanisms/mechanismsServer.Request()
Mar 24 12:41:34.932[31m [ERRO] [name:forwarder-vpp-2rx2r] [0m(13.1)               cannot support any of the requested mechanism
```

**Comment by denis-tingaikin** [comment]
This is nice to catch. Let's try to reproduce this via `sandbox` :)
Basically, as I understand we need to check these steps:

1. Deploy nsmgr
2. Deploy special forwarder that sets Payload
3. Check that Payload is not missing in the path.

**Comment by LionelJouin** [comment]
If the NSC is sending a request containing the network service endpoint name, the Payload will be missing in the path. So, in this case, the issue is still happening even with the last patch (#791).

**Comment by denis-tingaikin** [comment]
Will be fixed in #792
